### PR TITLE
extract evo scanning and processing events function from main to worker folder

### PR DESCRIPTION
**Pull request by zorjak**

### CODE REVIEW

File: internal/core/worker/evolution/worker.go [hunk 1]
```diff
@@ -0,0 +1,299 @@
...
+	"time"
+
+	"github.com/ethereum/go-ethereum/common"
+	"github.com/freeverseio/laos-universal-node/internal/config"
+	"github.com/freeverseio/laos-universal-node/internal/platform/model"
+	"github.com/freeverseio/laos-universal-node/internal/scan"
+	"github.com/freeverseio/laos-universal-node/internal/state"
+)
+
+type ReorgError struct {
```
Comment (luispdm) [inline internal/core/worker/evolution/worker.go]: Make it unexported

File: internal/core/worker/evolution/worker.go [hunk 2]
```diff
@@ -0,0 +1,299 @@
...
+			slog.Info("context canceled")
+			return nil
+		default:
+			lastBlock, err := getLastBlock(ctx, w.client, startingBlock, w.blocksRange, w.blocksMargin)
+			if err != nil {
+				slog.Error(err.Error())
+				break
+			}
+            if lastBlock < startingBlock {
+				slog.Debug("scanEvochain, last calculated block is behind starting block, waiting...",
```
Comment (luispdm) [inline internal/core/worker/evolution/worker.go]: Change `scanEvochain` to `evolution worker`

File: internal/core/worker/evolution/worker.go [hunk 3]
```diff
@@ -0,0 +1,299 @@
...
+	block       uint64
+	chainHash   common.Hash
+	storageHash common.Hash
+}
+
+func (e ReorgError) Error() string {
+	return "reorg error"
+}
+
+type Worker interface {
```
Comment (luispdm) [inline internal/core/worker/evolution/worker.go:28]: This interface isn't used anywhere

File: internal/core/worker/evolution/worker_test.go [hunk 4]
```diff
@@ -0,0 +1,596 @@
+package worker
```
Comment (luispdm) [inline internal/core/worker/evolution/worker_test.go:1]: As we test the private function `storeMintedWithExternalURIEventsByContract`, I'd rename the file `worker_wb_test.go`, the convention we have adopted so far to test private stuff (wb stands for white box)

**Review by luispdm — CHANGES_REQUESTED** [review]

File: internal/core/worker/evolution/worker.go (hunk 1 above)
Comment (zorjak) [reply internal/core/worker/evolution/worker.go]: ok

File: internal/core/worker/evolution/worker.go (hunk 3 above)
Comment (zorjak) [reply internal/core/worker/evolution/worker.go:28]: it is used used in main.go. when we run this worker in go rutine.

File: internal/core/worker/evolution/worker.go (hunk 2 above)
Comment (zorjak) [reply internal/core/worker/evolution/worker.go]: ok

File: internal/core/worker/evolution/worker_test.go (hunk 4 above)
Comment (zorjak) [reply internal/core/worker/evolution/worker_test.go:1]: if you are agree lets keep it as is. I am still thinking about moving things in process package and doing small refact. when I move tests there I will have clearer where this function should stay and where the test should be

Comment (luispdm) [reply internal/core/worker/evolution/worker_test.go:1]: As we're in the midst of a refactor, I'm ok with this for now

File: internal/core/worker/evolution/worker.go (hunk 3 above)
Comment (luispdm) [reply internal/core/worker/evolution/worker.go:28]: It is not, as `NewWorker` returns the private struct `worker`

Comment (zorjak) [reply internal/core/worker/evolution/worker.go:28]: ups. you are right. 

Fixed. NewWorker returns the interface now.

**Pull request merged**